packages/react-button/src/components/CompoundButton/useCompoundButton.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { mergeProps, resolveShorthandProps } from '@fluentui/react-compose/lib/next/index';
+import { CompoundButtonProps } from './CompoundButton.types';
+import { useButtonState } from '../Button/useButtonState';
+import { renderCompoundButton } from './renderCompoundButton';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const compoundButtonShorthandProps = ['icon', 'loader', 'contentWrapper', 'children', 'secondaryText'];
+
+/**
+ * Given user props, returns state and render function for a Button.
+ */
+export const useCompoundButton = (
+  props: CompoundButtonProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: CompoundButtonProps,
+) => {
+  // Ensure that the `ref` prop can be used by other things (like useFocusRects) to refer to the root.
+  // NOTE: We are assuming refs should not mutate to undefined. Either they are passed or not.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const resolvedRef = ref || React.useRef();
+  const state = mergeProps(
```

Comment:
isn't all of this done in useButtonState?

Reply:
It is done in `useButton` but not in `useButton` state.

Reply:
any reason to not do this in `useButtonState`?

Reply:
iirc we were doing this at this layer so that `useButtonState` was independent of our slots approach.

Reply:
I think this could just call useButton with only slightly different props

Reply:
@khmakoto - sorry i only meant to have the ref logic in useState hook, not the slots

Reply:
I'll address that in a separate PR, I want to get this one merged in :)